notifications_utils/field.py

```diff
@@ -75,20 +75,28 @@ def get_replacement(self, match):
         replacement = self.values.get(match.group(1) + match.group(3))
 
         if isinstance(replacement, list):
-            replacement = list(filter(None, replacement))
-            if not replacement:
-                return None
-            if self.markdown_lists:
-                return self.sanitizer('\n\n' + '\n'.join(
-                    '* ' + item for item in replacement
-                ))
-            return self.sanitizer(unescaped_formatted_list(replacement, before_each='', after_each=''))
-
-        if isinstance(replacement, str):
-            return self.sanitizer(replacement) or ''
+            return self.get_replacement_as_list(list(filter(None, replacement)))
+
+        if isinstance(replacement, bool):
+            return str(replacement)
+
+        if replacement:
```

Comment:
so if i've got this right here's the desired input/output for this:

```
None -> None
[] -> None

0 -> '0' (str)
'' -> '' (str)
bool -> 'bool' (str)

'truthy string' -> 'sanitized truthy str'
['truthy list'] -> get_replacement_as_list
```

don't think this'll catch the `0` case as it currently stands.